### PR TITLE
Graph_Implementation

### LINKED ISSUE #2052: Graph Implementation using HashMap- Java
There is no issue in code. I want to give an easy and better solution to Graph Implementation using HashMap in Java.

### PR DESCRIPTION
**Pull request by RitikaJain8818**
Fixes #2052 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
The Graph is implemented using HashMap in Java.

#### Languages Used:
Java

#### Files Added:
Graph_Implementation.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
complete pr template, squash commits, remove expression tree file from pr.

**Comment by RitikaJain8818** [comment]
Changes done.
Please check and let me know.

**Comment by jainaman224** [comment]
Incorrect issue number.

File: Graph/HashMap_Implementation_Graphs.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+
+class Graph {
+	private class Vertex {
```
Comment (jainaman224) [inline Graph/HashMap_Implementation_Graphs.java]: Use 4 space indentation.

File: Graph/HashMap_Implementation_Graphs.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		System.out.println("*****************");
+	}
+}
+
+class client {
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		Graph graph = new Graph();
+		graph.addVertex("A");
```
Comment (jainaman224) [inline Graph/HashMap_Implementation_Graphs.java]: Take input dynamically

**Comment by RitikaJain8818** [comment]
I have made the suggested changes.
Please check and let me know.

File: Graph/Graph_Implementation [hunk 3]
```diff
@@ -0,0 +1,158 @@
+import java.util.ArrayList;
```
Comment (jainaman224) [inline Graph/Graph_Implementation]: File extension is absent.

File: Graph/Graph_Implementation [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+import java.util.*;
+
+class Graph {
+	private class Vertex {
```
Comment (jainaman224) [inline Graph/Graph_Implementation]: Use 4 space indentation.

File: Graph/Graph_Implementation [hunk 5]
```diff
@@ -0,0 +1,158 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
```
Comment (jainaman224) [inline Graph/Graph_Implementation]: Change file name to Graph.java

**Comment by RitikaJain8818** [comment]
Changes Done.
Please check and let me know!

File: Graph/Graph_Implementation.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+class client {
+
+    public static void main(String[] args) {
+        Scanner scn=new Scanner(System.in);
+        // TODO Auto-generated method stub
+        Graph graph = new Graph();
+        int No_Of_Vertices=scn.nextInt();
+        while(No_Of_Vertices-->0){
+            String s=scn.next();
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Please add space around =.

File: Graph/Graph_Implementation.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            graph.addVertex(s);
+        }
+        int No_Of_Edges=scn.nextInt();
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Please add space around =.

File: Graph/Graph_Implementation.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        // edges are added to hashmap with weight
+        while(No_Of_Edges-->0){
+            String v1=scn.next();
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Please add space around =.

File: Graph/Graph_Implementation.java [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+            String v2=scn.next();
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Please add space around =.

File: Graph/Graph_Implementation.java [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+            int cost=scn.nextInt();
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Please add space around =.

File: Graph/Graph_Implementation.java [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+            graph.addEdge(v1, v2,cost);
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Please add space after a comma.

**Comment by somya-kapoor** [comment]
@RitikaJain8818 Are you working on it?

**Comment by RitikaJain8818** [comment]
> @RitikaJain8818 Are you working on it?

Yeah!

**Comment by RitikaJain8818** [comment]
Changes are done.
Please check

File: Graph/Graph_Implementation.java [hunk 12]
```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+import java.util.*;
+class Graph {
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+    }
+    public void display() {
+        Collection<Vertex> vtces = this.vtces.values();
+        for (Vertex vtx : vtces) {
+            vtx.display();
+        }
+        System.out.println("*****************");
+    }
+}
+class client {
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+    public static void main(String[] args) {
+        Scanner scn = new Scanner(System.in);
+        // TODO Auto-generated method stub
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+import java.util.*;
+class Graph {
+    private class Vertex {
+        String name;
+        HashMap<Vertex, Integer> nbrs;
+        Vertex(String name) {
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+            this.name = name;
+            this.nbrs = new HashMap<>();
+        }
+        public boolean equals(Object other) {
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+            Vertex ov = (Vertex) other;
+            return this.name.equals(ov.name);
+        }
+        public void display() {
+            String str = "";
+            str = this.name + "=>";
+            Set<Map.Entry<Vertex, Integer>> nbrs = this.nbrs.entrySet();
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+            for (Map.Entry<Vertex, Integer> nbr : nbrs) {
+                str = str + nbr.getKey().name + "(" + nbr.getValue() + "),";
+            }
+            str = str + "END";
+            System.out.println(str);
+        }
+    }
+    HashMap<String, Vertex> vtces;
+    public Graph() {
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 19]
```diff
@@ -0,0 +1,133 @@
...
+        this.vtces = new HashMap<>();
+    }
+    public int numvertices() {
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

File: Graph/Graph_Implementation.java [hunk 20]
```diff
@@ -0,0 +1,133 @@
...
+            return;
+        }
+        Vertex vtx = new Vertex(name);
+        this.vtces.put(name, vtx);
+    }
+    public void removeVertex(String name) {
+        if (!this.vtces.containsKey(name)) {
+            return;
+        }
+        Vertex vtx = this.vtces.get(name);
```
Comment (somya-kapoor) [inline Graph/Graph_Implementation.java]: Leave a line before it

**Comment by somya-kapoor** [comment]
@RitikaJain8818 Are you working on it?

**Comment by somya-kapoor** [comment]
@RitikaJain8818 Due to no activity for more than 7 days, I am closing this issue.

**Pull request closed without merging**